tonic-build/src/client.rs

```diff
@@ -34,6 +34,11 @@ pub(crate) fn generate(service: &Service, proto: &str) -> TokenStream {
                     Self { inner }
                 }
 
+                pub fn with_interceptor(inner: T, interceptor: impl Into<tonic::Interceptor>) -> Self {
+                    let inner = tonic::client::Grpc::with_interceptor(inner, interceptor);
+                    Self { inner }
+                }
+
```

Comment:
Do you think it would be worth it to use a builder-like pattern instead of `whit_interceptor` constructor? Same for the server.

Reply:
Since there is only one config option as of now, I wanted to keep this simple, I think if we add another option in the future we go with a builder.